bitwarden_license/src/Commercial.Core/SecretsManager/Commands/Secrets/CreateSecretCommand.cs

```diff
@@ -31,6 +31,11 @@ public async Task<Secret> CreateAsync(Secret secret, Guid userId)
             throw new NotFoundException();
```

Comment:
We should add a check that the project list only contains one project, or alter the access check to check all the projects provided.

It looks like it currently only checks the first project
```
    var hasAccess = accessClient switch
        {
            AccessClientType.NoAccessCheck => true,
            AccessClientType.User => await _projectRepository.UserHasWriteAccessToProject(project.Id, userId),
            _ => false,
        }
```

I think I prefer limiting to just one project in the secret's project list, since that is our intention currently.

That would simplify checking that the organizations all match. 

What do you think?

Reply:
I agree. I was thinking about making a different ticket for this, but I think we can just add it now. I've added a check at the controller level to ensure there is not more than one project for the Create and Update Secret endpoints. If we ever move towards a Secret in multiple projects we'll have to update things in multiple places. Let me know your thoughts on adding it there!